### ISSUE TITLE
Doc: Distinguish between elastic-supported and community-supported plugins

### ISSUE THREAD

**Issue by karenzone** [elastic/logstash#15970]
Can we use the `:default_plugin:` setting to accurately distinguish between elastic-supported and community-supported plugins.?

* Are all plugins that are installed by default elastic-supported plugins? 
* Are there any elastic-supported plugins that are not installed by default? 

~We can most likely implement this initiative by updating plugin headers to show different text based on the `:default_plugin:` value.~
Update: This approach won't work because there's not a 1:1 relationship between default plugins and level of support.   I'm trying something different as noted in comments. 

#### Proposal: Phased rollout
Follows the [Logstash Plugins support matrix](https://www.elastic.co/support/matrix#logstash_plugins). If we use this approach, let's consider a phased rollout:
- Tier 1 plugins (72 total)
- Tier 2 plugins (17 total)
- Community

#### Possible next steps
We could consider adding the configuration to `[plugins-metadata.json](https://github.com/elastic/logstash/blob/main/rakelib/plugins-metadata.json)`, but we'd still need to clear the setting in each doc source file. So there's probably no value in that approach.

**Comment by karenzone** [comment]
@flexitrev: Here's the issue we discussed. Please feel free to add to it. 

Who can answer these questions:
* Are all plugins that are installed by default elastic-supported plugins?
* Are there any elastic-supported plugins that are not installed by default?

**Comment by karenzone** [comment]
cc:/ @roaksoax @jsvd

**Comment by flexitrev** [comment]
I'm not sure we need to answer the default set of installed plugin question. Even if we do install an unsupported plugin or don't, that can change over time. What our documentation should reflect is what's currently supported. Or better, what versions are supported

**Comment by karenzone** [comment]
For sure, docs should reflect what's currently supported. I'm trying to figure out if/how we can use flags such as `default_plugin` to trigger conditional content rather than trying to manage it all manually.

**Comment by flexitrev** [comment]
@roaksoax would default_plugin work to distinguish supported/unsupported plugins?

**Comment by roaksoax** [comment]
@flexitrev @karenzone I don't think that that we can use `default_plugin` to differentiate whether a plugin is supported or unsupported. We do have unsupported plugins that are shipped by default in logstash, meaning they are set as `default_plugin`.

Either we expand the metadata to include that information, or we leverage the same mechanism as we do for the support matrix, which is to rely on the spreadsheet.

**Comment by karenzone** [comment]
@karenzone: Related: https://docs.elastic.co/integrations/support

**Comment by karenzone** [comment]
@karenzone: Related: https://docs.elastic.co/integrations/support

<img width="2341" alt="Screen Shot 2024-04-18 at 5 21 53 PM" src="https://github.com/elastic/logstash/assets/35154725/15843586-fc41-4b50-a1d3-597a52bef1a7">

**Comment by karenzone** [comment]
Note to self (@karenzone): Branched https://github.com/elastic/logstash-docs/pull/1672 to use as a test bed

**Comment by karenzone** [comment]
Related [test] PRs:
* #16206 
   Adds support for support level to plugin header files in `logstash` repo  
* https://github.com/elastic/logstash-docs/pull/1673
   Adds support level attributes to select generated plugin docs for validation

**Comment by karenzone** [comment]
@jsvd Let's discuss!

**Comment by jsvd** [comment]
I would like to avoid having more than 1 place where we state what's commercially supported.
The plugin reference docs explain how any/a user uses the plugin: if it needs to be installed or is bundled with logstash (:default-plugin: attr), what are its settings, and guidance for its use.
The support matrix is the place Elastic customers (current or prospect) can find out what is commercially supported if they have a subscription with Elastic. I'd prefer to keep this separation clean and avoid two sources of truth.

The Elastic integrations are, afaik, the only occurrence of the commercial support of a feature/product being defined outside of support.elastic.co.

**Comment by karenzone** [comment]
This was an ask from @flexitrev and @roaksoax in a move toward more transparency so that users would know what level of support to expect before they use a plugin.

**Comment by flexitrev** [comment]
This seems necessary to clear up current confusion about support for various plugins. Requiring customers to go to a different place to understand support than in documentation is a bad customer experience. I think being explicit about the plugins that are supported and therefore actively maintained is good practice.

**Comment by flexitrev** [comment]
I wanted to add here the prior art for this in elastic/integrations the owner is noted in the manifest https://github.com/elastic/integrations/blob/main/packages/akamai/manifest.yml

**Comment by karenzone** [comment]
Differences of opinion have stalled this work.  Closing for now.
If we decide to re-open, we have some PoC work already in place.
